openapi3/loader.go

```diff
@@ -693,12 +694,18 @@ func (loader *Loader) resolveSchemaRef(doc *T, component *SchemaRef, documentPat
 			}
 			component.Value = &schema
 		} else {
+			if visitedLimit(visited, ref, 3) {
+				visited = append(visited, ref)
+				return fmt.Errorf("invalid schema: circular schema reference detected - %s", strings.Join(visited, " -> "))
```

Comment:
Not an invalid schema but a bug in this lib! (that, by the way, I think has to do with `loader.visitedSchema` keeping track of `*Schema` but is missing the pointers to schemas that are embedded in a schema such as `items`, `not`, `oneOf`, `allOf` and the one that appears in the minified test of 542 `anyOf`. I'd dig that way :) )

I'm down to merge this safeguarding PR, just modify the error message to not say the document is invalid but that there is a bug with ongoing investigation. Introduce a package-editable var if you prefer.

If `3` is too small I'd be fine with `300`. As long as it returns before exhausting the stack and looping for too long, this PR is an improvement!

I'm interested in bringing https://github.com/APIs-guru/openapi-directory/tree/main/APIs into the CI test suite by the way. Let's finish this and bring this in another PR if that's fine with you :)

Reply:
updated the error message and made it configurable.

Also 3 I found was the sweet spot for the depth ( I didn't find any schemas that succeeded loading with 4 that couldn't load with 3 for example).